controlplane/kubeadm/internal/workload_cluster_conditions.go

```diff
@@ -253,22 +369,61 @@ func (w *Workload) UpdateStaticPodConditions(ctx context.Context, controlPlane *
 		allMachinePodConditions = append(allMachinePodConditions, controlplanev1.MachineEtcdPodHealthyCondition)
 	}
 
+	allMachinePodV1beta2Conditions := []string{
+		controlplanev1.KubeadmControlPlaneMachineAPIServerPodHealthyV1Beta2Condition,
+		controlplanev1.KubeadmControlPlaneMachineControllerManagerPodHealthyV1Beta2Condition,
+		controlplanev1.KubeadmControlPlaneMachineSchedulerPodHealthyV1Beta2Condition,
+	}
+	if controlPlane.IsEtcdManaged() {
+		allMachinePodV1beta2Conditions = append(allMachinePodV1beta2Conditions, controlplanev1.KubeadmControlPlaneMachineEtcdPodHealthyV1Beta2Condition)
+	}
+
 	// NOTE: this fun uses control plane nodes from the workload cluster as a source of truth for the current state.
+	// TODO: integrate this with clustercache / handle the grace period
 	controlPlaneNodes, err := w.getControlPlaneNodes(ctx)
 	if err != nil {
 		for i := range controlPlane.Machines {
 			machine := controlPlane.Machines[i]
 			for _, condition := range allMachinePodConditions {
 				conditions.MarkUnknown(machine, condition, controlplanev1.PodInspectionFailedReason, "Failed to get the node which is hosting this component: %v", err)
 			}
+
+			for _, condition := range allMachinePodV1beta2Conditions {
+				v1beta2conditions.Set(machine, metav1.Condition{
+					Type:    condition,
+					Status:  metav1.ConditionUnknown,
+					Reason:  controlplanev1.KubeadmControlPlaneMachinePodInspectionFailedV1Beta2Reason,
+					Message: fmt.Sprintf("Failed to get the Node hosting the %s", strings.TrimSuffix(condition, "Healthy")),
```

Comment:
TrimSuffix seems a bit brittle and also the resulting strings are looking a bit off. Do we want to use a clean switch instead to calculate the pod name?

Reply:
Dropped the Pod name entirely, we don't have it anywhere else.